py-polars/tests/unit/test_polars_import.py

```diff
@@ -67,4 +67,4 @@ def test_polars_import() -> None:
         assert isinstance(polars_import_time, int)
 
         if_err = f"Possible import speed regression; took {polars_import_time//1_000}ms"
-        assert polars_import_time < 200_000, f"{if_err}\n{df_import}"
+        assert polars_import_time < 400_000, f"{if_err}\n{df_import}"
```

Comment:
This is not acceptable - if you're noticing an impact on import time, it means the implementation does not utilize lazy loading correctly. This change should be reversed.

Reply:
I wasn't aware of this, let me fix this. I've reverted the change.

Reply:
I have to dig deeper into the lazy loading tomorrow

Reply:
I can also take a look if you want.

Reply:
Yes that would be nice, it is not directly obvious to me. It feels like I have to import from `polars.dependencies`, but that doesn't seem to work.

Reply:
> Yes that would be nice, it is not directly obvious to me. It feels like I have to import from `polars.dependencies`, but that doesn't seem to work.

Took a quick look; you had registered `pyiceberg` with the lazy loader, but then imported _directly_ from it at the head of the module, so it was promptly instantiated "for real". Looks fine now though - @stinodego beat me to it, heh ;)